src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -946,7 +939,7 @@ trait Contexts { self: Analyzer =>
           //     remedied nonetheless.
           Some(collectImplicits(owner.thisType.implicitMembers, owner.thisType))
         }
-      } else if (scope != nextOuter.scope && !owner.isPackageClass) {
+      } else if (scope != nextOuter.scope && !owner.isPackageClass && !(owner.isClass && owner.isInitialized)) {
```

Comment:
can we check whether this is local in one shot? Since `isClass` covers `isPackageClass`?